esp/esdllib/esdl_script.cpp

```diff
@@ -947,74 +1490,44 @@ class CEsdlCustomTransform : public CInterfaceOf<IEsdlCustomTransform>
 
 public:
     CEsdlCustomTransform(){}
-    void verifyPrefixDeclared(IPropertyTree &tree, const char *prefix)
-    {
-        StringBuffer attpath("@xmlns");
-        if (!isEmptyString(prefix))
-            attpath.append(':').append(prefix);
-        const char *uri = tree.queryProp(attpath.str());
-        if (!uri || !streq(uri, "urn:hpcc:esdl:script"))
-            throw MakeStringException(ESDL_SCRIPT_Error, "Undeclared script xmlns prefix %s", isEmptyString(prefix) ? "<default>" : prefix);
-    }
-    CEsdlCustomTransform(IPropertyTree &tree, const char *ns_prefix) : m_prefix(ns_prefix)
+
+    CEsdlCustomTransform(XmlPullParser &xpp, StartTag &stag, const char *ns_prefix) : m_prefix(ns_prefix)
     {
-        if (m_prefix.length())
-            m_prefix.append(':');
-        else
-        {
-            const char *tag = tree.queryName();
-            if (!tag)
-                m_prefix.set("xsdl:");
-            else
-            {
-                const char *colon = strchr(tag, ':');
-                if (!colon)
-                    verifyPrefixDeclared(tree, nullptr);
-                else
-                {
-                    if (colon == tag)
-                        throw MakeStringException(ESDL_SCRIPT_Error, "Tag shouldn't start with colon %s", tag);
-                    m_prefix.append(colon-tag, tag);
-                    if (!streq(m_prefix, "xsdl"))
-                        verifyPrefixDeclared(tree, m_prefix);
-                    //add back the colon for easy comparison
-                    m_prefix.append(':');
-                }
-            }
-        }
+        const char *tag = stag.getLocalName();
 
-        m_name.set(tree.queryProp("@name"));
-        m_target.set(tree.queryProp("@target"));
+        m_name.set(stag.getValue("name"));
+        m_target.set(stag.getValue("target"));
```

Comment:
Since we can use @target as a shortcut in place of a target element, and we have to for legacy, does it make sense to provide a corresponding shortcut for the source document?

Reply:
Yes. I had thought about that, and it's a good idea.  Not sure if it will make it in this PR or not, but if not I'll open a JIRA.

Reply:
done.